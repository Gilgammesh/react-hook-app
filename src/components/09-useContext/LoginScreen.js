import React, { useContext } from "react";
import { useHistory } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import NavBar from "./NavBar";
import UserContext from "./UserContext";

const LoginScreen = () => {
  const history = useHistory();

  const initialForm = {
    id: "",
    name: "",
    email: "",
    password: "",
  };
  const [formValues, handleInputChange, resetForm] = useForm(initialForm);
  const { name, email, password } = formValues;

  const { setUser } = useContext(UserContext);

  const handleSubmit = (evt) => {
    evt.preventDefault();
    setUser({ ...formValues, id: Math.floor(Math.random() * 100) });
    resetForm(initialForm);
    history.push("/home");
  };

  return (
    <>
      <NavBar />
      <div className="m-4">
        <h1>LoginScreen</h1>
        <hr />
        <form onSubmit={handleSubmit}>
          <div className="form-group">
            <label htmlFor="name">Nombre</label>
            <input
              type="text"
              className="form-control"
              id="name"
              name="name"
              value={name}
              onChange={handleInputChange}
              required
            />
          </div>
          <div className="form-group">
            <label htmlFor="email">Email</label>
            <input
              type="email"
              className="form-control"
              id="email"
              name="email"
              value={email}
              onChange={handleInputChange}
              required
            />
          </div>
          <div className="form-group">
            <label htmlFor="password">Contrase√±a</label>
            <input
              type="password"
              className="form-control"
              id="password"
              name="password"
              value={password}
              onChange={handleInputChange}
              required
            />
          </div>
          <button type="submit" className="btn btn-primary">
            Ingresar
          </button>
        </form>
      </div>
    </>
  );
};

export default LoginScreen;
