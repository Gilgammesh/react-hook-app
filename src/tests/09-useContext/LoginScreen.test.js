import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow, mount } from "enzyme";
import "@testing-library/jest-dom";
import LoginScreen from "../../components/09-useContext/LoginScreen";
import UserContext from "../../components/09-useContext/UserContext";

describe("Pruebas unitarias en el componente LoginScreen.js", () => {
  const user = {
    id: 1,
    name: "Carlos",
    email: "carlos.santander@gmail.com",
    password: "123456",
  };
  const setUser = jest.fn();
  /************************************************************************************/
  test("se debe mostrar el componente <LoginScreen /> correctamente", () => {
    const wrapper = mount(
      <UserContext.Provider
        value={{
          user,
          setUser,
        }}
      >
        <Router>
          <LoginScreen />
        </Router>
      </UserContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe ejecutar el setUser con el argumento esperado", () => {
    const wrapper = mount(
      <UserContext.Provider
        value={{
          user,
          setUser,
        }}
      >
        <Router>
          <LoginScreen />
        </Router>
      </UserContext.Provider>
    );

    wrapper.find("input#name").prop("onChange")({
      target: {
        type: "text",
        checked: false,
        name: "name",
        value: user.name,
      },
    });
    wrapper.find("input#email").prop("onChange")({
      target: {
        type: "text",
        checked: false,
        name: "email",
        value: user.email,
      },
    });
    wrapper.find("input#password").prop("onChange")({
      target: {
        type: "text",
        checked: false,
        name: "password",
        value: user.password,
      },
    });
  });
});
