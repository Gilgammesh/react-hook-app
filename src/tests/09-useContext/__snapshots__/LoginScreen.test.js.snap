// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente LoginScreen.js se debe mostrar el componente <LoginScreen /> correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <div
      className="navbar-collapse"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </ul>
    </div>
  </nav>,
  <div
    className="m-4"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="name"
        >
          Nombre
        </label>
        <input
          className="form-control"
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Contrase√±a
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Ingresar
      </button>
    </form>
  </div>,
]
`;
